kvdb-rocksdb/src/lib.rs

```diff
@@ -680,6 +681,15 @@ impl Database {
 			HashMap::new()
 		}
 	}
+
+	/// Try to catch up a secondary instance of the database with
+	/// the primary instance by reading as much from the logs as possible
+	pub fn try_catch_up_with_primary(&self) -> io::Result<()> {
```

Comment:
What happens if I misuse this and call it as a primary?

Reply:
It returns an error that originating from the C implementation of RocksDB. I'm unfamiliar with how these errors should be handled, however. It is currently transformed into a io error which seems like how the rest of the code handles it. Should I add a note about misuse?

Reply:
IO error is fine imo; adding a note saying "Calling this as primary will return an error" is Good Enough™.